semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/BindingMethodOverridesPass.scala

```diff
@@ -1,48 +1,54 @@
 package io.shiftleft.semanticcpg.passes
 
 import io.shiftleft.codepropertygraph.Cpg
-import io.shiftleft.codepropertygraph.generated.{NodeKeys, nodes}
-import io.shiftleft.passes.{CpgPass, DiffGraph, ParallelIteratorExecutor}
+import io.shiftleft.codepropertygraph.generated.{nodes, NodeKeyNames}
+import io.shiftleft.passes.{CpgPass, DiffGraph}
 import io.shiftleft.semanticcpg.language._
-import org.apache.logging.log4j.LogManager
+import scala.jdk.CollectionConverters._
 
 class BindingMethodOverridesPass(cpg: Cpg) extends CpgPass(cpg) {
-  import BindingMethodOverridesPass._
+  val overwritten = scala.collection.mutable.HashSet[nodes.Binding]()
+  val bindingTable = scala.collection.mutable.HashMap[(String, String, nodes.TypeDecl), nodes.Binding]()
 
   override def run(): Iterator[DiffGraph] = {
-    val methodIterator = cpg.method.toIterator()
-    new ParallelIteratorExecutor(methodIterator).map(processMethod)
-  }
-
-  private def processMethod(method: nodes.Method): DiffGraph = {
-    val diff = DiffGraph.newBuilder
-    method.start.referencingBinding
-      .toIterator()
-      .foreach(binding => {
-        val typeDecl: Option[nodes.TypeDecl] = binding.start.bindingTypeDecl.headOption()
-        if (typeDecl.isDefined) {
-          val neverOverriddenFlag = isMethodNeverOverridden(typeDecl.get, binding.name, binding.signature)
-          diff.addNodeProperty(
-            binding,
-            NodeKeys.IS_METHOD_NEVER_OVERRIDDEN.name,
-            neverOverriddenFlag.asInstanceOf[AnyRef]
-          )
-        } else {
-          logger.error(
-            "No binding typeDecl found in BindingMethodOverridesPass: " + method.name + " " + method.signature)
+    val diffGraph = DiffGraph.newBuilder
+    for (typeDecl <- cpg.typeDecl.toIterator;
+         binding <- typeDecl.bindsOut.asScala) {
+      bindingTable.update((binding.name, binding.signature, typeDecl), binding)
+    }
+    for (typeDecl <- cpg.typeDecl.toIterator) {
+      val parentTypeDecls = typeDecl.inheritsFromOut.asScala.map {
+        _.refOut.next
+      }.toList
+      for (binding <- typeDecl.bindsOut.asScala) {
+        if (!overwritten.contains(binding)) {
+          val method = binding.refOut.next
+          for (parentTypeDecl <- parentTypeDecls) {
+            val parentBinding = bindingTable.get((binding.name, binding.signature, parentTypeDecl))
+            if (parentBinding.isDefined && parentBinding.get.refOut.next != method) {
```

Comment:
```suggestion
            parentBinding.foreach { ... }
```
At least that way don't have to spell out both `isDefined` and `get`.

Reply:
Afaiu `foreach` opens a stackframe / closure. That's more complex and therefore harder to read than a simple conditional branch.